internal/sleuth/project_resource.go

```diff
@@ -0,0 +1,248 @@
+package sleuth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/sleuth-io/terraform-provider-sleuth/internal/gqlclient"
+)
+
+var (
+	_ resource.Resource                = &projectResource{}
+	_ resource.ResourceWithConfigure   = &projectResource{}
+	_ resource.ResourceWithImportState = &projectResource{}
+)
+
+type projectResourceModel struct {
+	Name                      types.String `tfsdk:"name"`
+	Slug                      types.String `tfsdk:"slug"`
+	Description               types.String `tfsdk:"description"`
+	IssueTrackerProviderType  types.String `tfsdk:"issue_tracker_provider_type"`
+	BuildProvider             types.String `tfsdk:"build_provider"`
+	ChangeFailureRateBoundary types.String `tfsdk:"change_failure_rate_boundary"`
+	ImpactSensitivity         types.String `tfsdk:"impact_sensitivity"`
+	FailureSensitivity        types.Int64  `tfsdk:"failure_sensitivity"`
+}
+
+type projectResource struct {
+	c *gqlclient.Client
+}
+
+func NewProjectResource() resource.Resource {
+	return &projectResource{}
+}
+
+func (p *projectResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Project resource manages Sleuth project.",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Project name",
+				Required:            true,
+			},
+			"slug": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Project slug",
+			},
+			"description": schema.StringAttribute{
+				DeprecationMessage:  "Project description will be removed",
+				MarkdownDescription: "Project description",
+				Optional:            true,
+				Computed:            true,
+				Default:             stringdefault.StaticString(""),
```

Comment:
If default is present, then attribute must be computed

Reply:
Why is that?

Reply:
I belive it's because of how new workflow computes - so if there is a value present in *.tf file it will take it if not it will "compute" it from this Default function. I am not completely sure TBH, but this was the feeling when I read the code because their docs are not really detailed on that matter.

Reply:
I'm not sure that's the case based on the code and docs [here](https://github.com/hashicorp/terraform-plugin-framework/blob/main/resource/schema/string_attribute.go#L148-L154).

Reply:
Actually just a few lines below you can see it validates that specifically: https://github.com/hashicorp/terraform-plugin-framework/blob/cf1f3486e55faa46001d62433db3b5af0260e350/resource/schema/string_attribute.go#L232-L240

Reply:
But yeah my guess is that they kept the same API as it was before and have just added these specifics inside.

Reply:
That looks to be true. Weird. I can think of multiple examples of using Terraform in the past where I wanted a default, but it wasn't a computed value. 🤷🏼 Fair enough.